src/terrain/builder.cpp

```diff
@@ -824,7 +824,7 @@ void terrain_builder::parse_mapstring(
 			} else if(terrain.overlay != 0) {
 				anchors.emplace(terrain.overlay, map_location(x, y));
 			} else if(terrain.base == t_translation::TB_STAR) {
-				add_constraints(br.constraints, map_location(x, y), t_translation::STAR, global_images);
+				add_constraints(br.constraints, map_location(x, y), t_translation::ter_match(t_translation::STAR), global_images);
```

Comment:
I'm not quite sure what's happening here, but would it make sense for `t_translation::STAR` to be of type `t_translation::ter_match`?